compose/compose-file/compose-versioning.md

```diff
@@ -0,0 +1,301 @@
+---
+description: Compose file reference
+keywords: fig, composition, compose, versions, upgrading, docker
+title: Compose file versions and upgrading
+---
+
+The Compose file is a [YAML](http://yaml.org/) file defining services,
+networks, and volumes for a Docker application.
+
+There are several versions of the Compose file reference available.
+
+- [Version 3](index.md) (most current, and recommended)
+
+- [Version 2 and Version 2.1](compose-file-v2.md)
+
+- [Version 1](compose-file-v1.md)
+
+The topics below explain the differences among the versions, Docker Engine compatibility, and how to upgrade.
+
+## Compatibility matrix
+
+{% include content/compose-matrix.md %}
+
+## Versioning
+
+There are currently four versions of the Compose file format:
+
+- Version 1, the legacy format. This is specified by
+omitting a `version` key at the root of the YAML.
+
+- Version 2. This is specified with a `version: '2'`
+entry at the root of the YAML.
+
+- Version 2.1, an upgrade over version 2 that takes
+advantage of newer Docker Engine features. Specify with a
+`version: '2.1'` entry at the root of the YAML.
+
+- Version 3, the latest and recommended version, designed to
+be cross-compatible between Compose and the Docker Engine's
+[swarm mode](/engine/swarm/index.md).
```

Comment:
Can we group 2 and 2.1 together? So it's "Version 2.x ....", and Version 3.x" ....

That way we don't have to make multiple changes to these docs on every minor version release, we can just add the fields to the latest reference page.